### PR TITLE
`IntegrationTests`: simplified tests using `async` APIs

### PR DESCRIPTION
**Pull request by NachoSoto**
### Improvements:
- No more `expect(completed).toEventually(beTrue())`.
- No more `expect(error).to(beNil())`. Using `try` and `throws` tests.
- No more `expect(result).toNot(beNil())` or `XCTUnwrap`. The new APIs already return non-optional data when not failed.
- No more `toEventually`s + polling. Tests run serially now without busy waiting, which will make them faster.
- No more completion block trees, tests are linear now.

### Other changes:
- Removed empty optional `purchases(_:, shouldPurchasePromoProduct:, defermentBlock:)`.
- Using modern `==` Nimble methods.
- Removed redundant `syncPurchases` call when purchasing.

### CODE REVIEW

File: BackendIntegrationTests/BackendIntegrationTests.swift [hunk 1]
```diff
@@ -63,266 +60,157 @@ class BackendIntegrationSK1Tests: XCTestCase {
...
-            receivedOfferings = offerings
-        }
-        expect(completionCalled).toEventually(beTrue(), timeout: Self.timeout)
-
-        expect(receivedError).to(beNil())
-        let unwrappedOfferings = try XCTUnwrap(receivedOfferings)
-        expect(unwrappedOfferings.all).toNot(beEmpty())
+    func testCanGetOfferings() async throws {
+        let receivedOfferings = try await Purchases.shared.offerings()
+        expect(receivedOfferings.all).toNot(beEmpty())
```
Comment (NachoSoto) [inline BackendIntegrationTests/BackendIntegrationTests.swift:65]: This test is now only 2 lines 🤪

Comment (aboedo) [reply BackendIntegrationTests/BackendIntegrationTests.swift:65]: I think I might cry

File: BackendIntegrationTests/BackendIntegrationTests.swift [hunk 2]
```diff
@@ -63,266 +60,157 @@ class BackendIntegrationSK1Tests: XCTestCase {
...
 
-                expect(created).to(beTrue())
-                expect(identifiedCustomerInfo?.entitlements["premium"]?.isActive) == true
-                completionCalled = true
-            }
-        }
-        expect(completionCalled).toEventually(beTrue(), timeout: Self.timeout)
+        let (identifiedCustomerInfo, created) = try await Purchases.shared.logIn(identifiedUserID)
+        expect(created) == true
+        expect(identifiedCustomerInfo.entitlements["premium"]?.isActive) == true
```
Comment (aboedo) [inline BackendIntegrationTests/BackendIntegrationTests.swift:88]: this is so easy to read now

File: BackendIntegrationTests/BackendIntegrationTests.swift [hunk 3]
```diff
@@ -63,266 +60,157 @@ class BackendIntegrationSK1Tests: XCTestCase {
...
-            }
-        }
+        var (_, created) = try await Purchases.shared.logIn(identifiedUserID)
+        expect(created) == true
 
-        expect(completionCalled).toEventually(beTrue(), timeout: Self.timeout)
+        _ = try await Purchases.shared.logOut()
+
+        (_, created) = try await Purchases.shared.logIn(identifiedUserID)
+        expect(created) == false
```
Comment (aboedo) [inline BackendIntegrationTests/BackendIntegrationTests.swift:139]: I cannot overstate my joy

File: BackendIntegrationTests/BackendIntegrationTests.swift [hunk 4]
```diff
@@ -63,266 +60,157 @@ class BackendIntegrationSK1Tests: XCTestCase {
...
-            }
-
-            DispatchQueue.main.asyncAfter(deadline: .now() + .seconds(2)) {
-                Purchases.shared.syncPurchases(completion: completion)
-            }
-        }
+        return try await Purchases.shared
+            .purchase(package: monthlyPackage)
+            .customerInfo
     }
```
Comment (aboedo) [inline BackendIntegrationTests/BackendIntegrationTests.swift:214]: 👏👏👏👏

**Review by aboedo — APPROVED** [review]
this refactor made my week 🏆

**Pull request merged**